packages/backend/src/core/balances/BalanceUpdater.ts

```diff
@@ -39,6 +40,7 @@ export class BalanceUpdater {
     private readonly clock: Clock,
     private readonly projects: BalanceProject[],
     private readonly logger: Logger,
+    private readonly chainId: ChainId,
```

Comment:
Hmm, I'm thinking if actually checking the chainId against other services (multicall in particular here) wouldn't be a good idea here. Or maybe even the multicall can hold this information? This would make sure thet we're downloading the data from the correct chain.